File: ria/java/gof/src/main/java/com/gof/pattern/abstractFactory/after/ShipInventory.java
package com.gof.pattern.abstractFactory.after;

import com.gof.pattern.factoryMethod.after.Ship;
import com.gof.pattern.factoryMethod.after.ShipFactory;

public class ShipInventory {
    public static void main(String[] args) {
//        ShipFactory shipFactory = new WhiteShipFactory(new WhitePartsProFactory());
        ShipFactory shipFactory = new WhiteShipFactory(new WhiteShipPartsFactory());
        Ship ship = shipFactory.createShip();
        System.out.println("ship.getAnchor().getClass() = " + ship.getAnchor().getClass());
        System.out.println("ship.getWheel().getClass() = " + ship.getWheel().getClass());
    }
}

File: ria/gof/src/main/kotlin/com/gof/pattern/abstractFactory/after/ShipInventory.kt
package com.gof.pattern.abstractFactory.after

import com.gof.pattern.factory.Ship
import com.gof.pattern.factory.ShipFactory

class ShipInventory

fun main() {
    val shipFactory: ShipFactory = WhiteShipFactory(WhitePartsProFactory())
    val ship: Ship = shipFactory.createShip()
    println("ship.anchor.class() = ${ship.anchor!!::class}")
    println("ship.wheel.class() = ${ship.wheel!!::class}")
}
